packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -272,6 +298,13 @@ export class WaferMap extends FoundationElement {
 
     private diesChanged(): void {
         this.waferMapUpdateTracker.track('dies');
+        this.renderStrategy = this.dies === undefined ? 'worker' : 'main';
+        this.waferMapUpdateTracker.queueUpdate();
+    }
+
+    private diesTableChanged(): void {
+        this.waferMapUpdateTracker.track('dies');
+        this.renderStrategy = this.diesTable === undefined ? 'main' : 'worker';
         this.waferMapUpdateTracker.queueUpdate();
     }
```

Comment:
This seems kind of racy. Property binding should behave as declaratively as possible. It's weird that assigning these in different orders will result in different behavior. Can we instead say that if diesTable is defined use that for both of these?

```ts
if (this.diesTable) {
    return 'worker';
}
return 'main';
```

Also is having the string representation actually useful?

Seems like an unnecessary extra abstraction. Can we just do

```ts
    public readonly mainRenderer = new RenderingModule(this);
    public readonly workerRenderer = new WorkerRenderer(this);

    @observable
    public renderer = this.mainRenderer;
```